finality-aleph/src/network/gossip/mod.rs

```diff
@@ -15,18 +16,34 @@ mod service;
 pub use service::Service;
 
 #[async_trait::async_trait]
-/// Interface for the gossip network, currently only supports broadcasting and receiving data.
+/// Interface for the gossip network. This represents a P2P network and a lot of the properties of
+/// this interface result from that. In particular we might know the ID of a given peer, but not be
+/// connected to them directly.
 pub trait Network<D: Data>: Send + 'static {
     type Error: Display + Send;
+    type PeerId: Clone + Debug + Eq + Hash + Send + 'static;
```

Comment:
Maybe we could just reuse `PeerId` trait here from `finality-aleph/src/network/mod.rs`?

Reply:
Eh, there are some differences between these traits, I would prefer not to unify them.